### PR TITLE
Server delete poll

### PR DESCRIPTION
**Pull request by TimSimpsonR**
This fixes pagination by fixing our delete stuff.

### CODE REVIEW

File: reddwarf/instance/models.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
 from reddwarf.common.remote import create_guest_client
 from reddwarf.common.remote import create_nova_client
 from reddwarf.common.remote import create_nova_volume_client
+from reddwarf import db as db_api
```
Comment (ed-) [inline reddwarf/instance/models.py]: My fault but I think with the find_all and find_by methods no longer being overridden this import isn't needed

**Comment by ed-** [comment]
+1

File: reddwarf/taskmanager/models.py [hunk 2]
```diff
@@ -202,6 +202,28 @@ def delete_instance(self):
...
+            except nova_exceptions.NotFound:
+                return True
+
+        try:
+            poll_until(server_is_finished, sleep_time=2,
+                    time_out=int(config.Config.get('server_delete_time_out')))
+        finally:
+            LOG.debug("Setting instance %s to deleted..." % self.id)
+            self.update_db(task_status=InstanceTasks.NONE)
+            #TODO: Set deleted.
```
Comment (hub-cap) [inline reddwarf/taskmanager/models.py]: Why this comment?

Comment (rnirmal) [inline reddwarf/taskmanager/models.py]: I suppose we can get rid of this todo :)

File: reddwarf/taskmanager/models.py [hunk 3]
```diff
@@ -202,6 +202,28 @@ def delete_instance(self):
...
+                        "of instance %s!" % (server.id, self.id))
+                    return True
+                return False
+            except nova_exceptions.NotFound:
+                return True
+
+        try:
+            poll_until(server_is_finished, sleep_time=2,
+                    time_out=int(config.Config.get('server_delete_time_out')))
+        finally:
```
Comment (hub-cap) [inline reddwarf/taskmanager/models.py]: So i guess even if it fails in the time_out we set it to deleted? why not just set it to deleted w/o a poll then :)

**Comment by cp16net** [comment]
+1

File: reddwarf/instance/models.py (hunk 1 above)
Comment (TimSimpsonR) [inline reddwarf/instance/models.py]: Fixed.

File: reddwarf/taskmanager/models.py (hunk 2 above)
Comment (TimSimpsonR) [inline reddwarf/taskmanager/models.py]: Done.

**Comment by cp16net** [comment]
+1

File: reddwarf/taskmanager/models.py (hunk 3 above)
Comment (TimSimpsonR) [inline reddwarf/taskmanager/models.py]: Changed this. Now if the poll_until times out or the server gets into an error state, no delete=True for you.

**Comment by rnirmal** [comment]
+1

**Comment by hub-cap** [comment]
nice +1

**Pull request merged**